pkg/render/common/components/components.go

```diff
@@ -294,6 +302,20 @@ func applyReplicatedPodResourceOverrides(r *replicatedPodResource, overrides any
 	return r
 }
 
+// For UT purposes, only, this variable stores the override fields that were handled in that most
+// recent `applyReplicatedPodResourceOverrides` call.  UT code then checks that the structures we
+// use for `overrides` do not have any _other_ fields than those (except for known special cases).
+var overrideFieldsHandledInLastApplyCall []string
```

Comment:
I'm fine for leaving this, but one problem with this approach is that if we do change the tests to be the component_test package then this becomes a little more difficult to test.

For the most part I like using the _test package because it forces the uts to only interact with the interface that is accessible outside of the package and therefore test the same code paths that are used by the users of the package.

The operator project doesn't really do this in general though, and I think addressing this change would be the least of the issues we'd have to deal with if we were to start doing this.

Ultimately, seems like there's no awesome way to deal with this without possibly changing the interfaces we're testing, and probably not worth it to spend any more time iterating on this.

Reply:
Thanks, Brian, I agree in principle with the concerns here.  But as you say, I don't currently see any better solution.  I could add an optional reporter arg to `applyReplicatedPodResourceOverrides`, but then I'd also need to pass that to all of the methods like `GetMetadata` - which feels like a bit too invasive, e.g. there would be an inscrutable `, nil` added to all of the calls to those methods from the validation code.  Or we could make `applyReplicatedPodResourceOverrides` etc. methods of some override management component - but that would be a bigger change.  I think what we have is a decent balance for now.